### PR TITLE
feat: Integrate cluster-info-client, automatically fetching and storing cluster info

### PR DESCRIPTION
**Pull request by aearly**
This creates a new command, `create`, which can prevision a cluster using a minimum of input.  It will pull defaults and/or exisitng cluster data from vault, provision a server, and store the results in vault without any config hitting disk.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,6 @@ node_modules/
 package-lock.json
 gs-auth.json
 gs-test-meta.js
-cluster-*.json
+*cluster*.json
+seed.js
```
[the remark this answers is not in the document]
Comment (nlf) [reply .gitignore]: is it a thing we'll use again? maybe keeping a template with fake values around and documenting its usage is a good idea

File: README.md [hunk 2]
```diff
@@ -83,8 +83,32 @@ It is recommended that sensitive data (like the Kubernetes admin password) is st
 
 A CLI is also provided for `fabrik8` that allows you to invoke the API from the command line:
 
+### `fabrik8 create [--name name] [--url url] --spec ./path/to/spec`
+
+Creates a full cluster, reading defaults and existing configuration securely from centralized cluster-info.  The only options that are required are configuration for cluster-info, a name or cluster url, and the path to a McGonagall specification.  If re-running, only a name is required -- options will be re-read from cluster-info.
+
+* `--url`, `-u` the url of the cluster you wish to create,
```
Comment (nlf) [inline README.md]: nit: an example value here would be nice, url is a little vague (does it include the registry prefix, the www prefix, the protocol, the trailing slash, etc etc)

File: spec/api.spec.js [hunk 3]
```diff
@@ -198,8 +218,8 @@ describe('API', function () {
           return Promise.resolve(specData)
         }
 
-        let tempSpec = { data: { ...specData, cluster: clusterDetail }, ...options }
-        clusterInfo = Object.assign({}, clusterDetail, { specData: tempSpec })
+        let tempSpec = { ...specData, ...options, masterIP: '192.168.1.1', credentials: SERVICE_ACCOUNT }
```
Comment (nlf) [inline spec/api.spec.js]: just to verify my assumption, masterIP in this context is a fake value and not intended to point to anything real?

File: src/api.js [hunk 4]
```diff
@@ -1,14 +1,20 @@
 const bole = require('bole')
 const log = bole('fabrik8.api')
-const fount = require('fount')
 const filterUndefined = require('./filter')
 
-async function initialize (events, Kubeform, hikaru, clusterConfig, specification, data = {}, options = { data }) {
+const initialize = (events, Kubeform, hikaru) => async (kubeformParams, specification, hikaruParams, options = {}) => {
```
[the remark this answers is not in the document]
Comment (nlf) [reply src/api.js]: i like this! much clearer

File: src/commands/create.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
+const createInfoClient = require('@npm-wharf/cluster-info-client')
+const createReconciler = require('../reconcile')
+const bole = require('bole')
+
+function build () {
+  return {
+    url: {
+      description: 'the url of the cluster you wish to create',
```
Comment (nlf) [inline src/commands/create.js]: the example value in the readme should probably be here somewhere too

File: src/commands/init.js [hunk 6]
```diff
@@ -147,7 +147,7 @@ function writeOutput (result, argv) {
 module.exports = function (fabricator, debugOut) {
   return {
     command: 'init <source> [options]',
-    desc: 'performs full provisioning of a Kubernetes cluster and deployment of software',
+    desc: '[DEPRECATED, use `create`] performs full provisioning of a Kubernetes cluster and deployment of software',
```
Comment (nlf) [inline src/commands/init.js]: :clap: :clap: :clap:

File: src/reconcile.js [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+    _reifyServiceAccounts
+  }
+
+  /**
+   * A lot is going on here!  We have to take in the yargs args, and then
+   * produce the params needed for fabrik8.  Some of these will be inferred
+   * from the input, others will come from Vault as default values.  Some params
+   * will come from previous fabrik8 runs, if the cluster exists!  We also have
+   * to process args that override defaults, and generate certain missing values
+   * according to rules specified in the common default config.
```
Comment (nlf) [inline src/reconcile.js]: having all of this in one place is _awesome_ so thank you for that

i'd love to have some kind of documentation as to what all of the parameters are and what sources they can come from, as well as how other sources fit in order of precedence, but that is very much not in scope for this pull request and should not be a blocker

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (aearly) [reply .gitignore]: Good idea. I checked seed.js in, with some scrubbed defaults.

File: spec/api.spec.js (hunk 3 above)
Comment (aearly) [reply spec/api.spec.js]: Yep, totally fake for testing.

**Pull request merged**